streams/runners/storm/layout/src/main/java/com/hortonworks/streamline/streams/layout/storm/KafkaSpoutFluxComponent.java

```diff
@@ -113,15 +116,49 @@ private String addSpoutConfigComponent () {
 
     private String addSchemeComponent() {
         String streamsSchemeId = "streamsScheme-" + UUID_FOR_COMPONENTS;
-        String schemeClassName = "com.hortonworks.streamline.streams.runtime.storm.spout.AvroKafkaSpoutScheme";
-        String topicName = (String) conf.get(TopologyLayoutConstants.JSON_KEY_TOPIC);
-        String schemaRegistryUrl = (String) conf.get(TopologyLayoutConstants.SCHEMA_REGISTRY_URL);
-        List<String> constructorArgs = Lists.newArrayList((kafkaSource != null ? kafkaSource.getId() : ""), topicName, schemaRegistryUrl);
-        addToComponents(createComponent(streamsSchemeId, schemeClassName, null, constructorArgs, null));
+        String schemeClassName = (String) conf.getOrDefault(TopologyLayoutConstants.JSON_KEY_SCHEME_CLASS_NAME,
+                                                            DEFAULT_SCHEME_CLASS);
+        if(schemeClassName.isEmpty()) {
+            throw new RuntimeException("Property [" + TopologyLayoutConstants.JSON_KEY_SCHEME_CLASS_NAME
+                                               + "] value can not be an empty String, it must be a valid class.");
+        }
+
+        Map<String, Object> configMethod = getConfigMethodWithRefArgs("init",
+                                                                      Collections.singletonList(addConfigMapInstance()));
+        addToComponents(createComponent(streamsSchemeId,
+                                        schemeClassName,
+                                        null,
+                                        null,
+                                        Collections.singletonList(configMethod)));
 
         return streamsSchemeId;
     }
 
+    private String addConfigMapInstance() {
+        String mapClassId = "map-" + UUID_FOR_COMPONENTS;
+        List<Pair<String, Object>> entries =
+                Lists.newArrayList(Pair.of(KafkaSourceScheme.TOPIC_KEY,
+                                           conf.get(TopologyLayoutConstants.JSON_KEY_TOPIC)),
+                                   Pair.of(KafkaSourceScheme.SCHEMA_REGISTRY_URL_KEY,
+                                           conf.get(TopologyLayoutConstants.SCHEMA_REGISTRY_URL)),
+                                   Pair.of(KafkaSourceScheme.DATASOURCE_ID_KEY,
+                                           kafkaSource != null ? kafkaSource.getId() : ""));
+        List<Map<String, Object>> configMethods = new ArrayList<>();
+        for (Map.Entry<String, Object> entry : entries) {
```

Comment:
Initially I had that code and it did not look good to me. Avoided multiple statements to set configMethods by having a list with 3 elements.